### PR TITLE
bind message gateway id

### LINKED ISSUE #4269: send MessageID to FES for easier threading of conversations on Gmail
Roman - on FES we allowed two methods to supply message ID (the mime email header).

Option 1: pre-generate message id before constructing mime / uploading the message to Gmail, and include that in FES /upload call

Option 2: let Gmail generate message id (as it does now) and then retrieve it & send that to FES in another API request

Option 1 is preferred if it can be made to work (if Gmail respects message id constructed by us when we're sending the message) as it requires fewer API calls.

Pre-generate option:
- create a valid Message-ID - these should include the angled brackets.
- Send it along in the MessageUploadRequest class:
```kotlin
data class MessageUploadRequest(
   val associateReplyToken: String,
   val expiration: Instant?,
   val from: String,
   val to: List<String>,
   val cc: List<String> = emptyList(),
   val bcc: List<String> = emptyList(),
   val emailGatewayMessageId: String? = null
)
```

Separate API request option:
- after sending the email retrieve the Message-ID.
- Send it to the new POST
/api/$VERSION_CLIENT/message/{externalId}/gateway endpoint.
Authorization is required and the class is:
```kotlin
data class MessageGatewayUpdateRequest(
   val emailGatewayMessageId: String,
)
```

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
This PR binds message gateway id for pwd-encrypted messages

close #4269

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by rrrooommmaaa** [comment]
> Sorry for lack of info. Indeed, I suppose we didn't update FES yet. I'll let you know once updated.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-browser/issues/4284#issuecomment-1077716906_

Should I keep this PR draft meanwhile?

**Comment by tomholub** [comment]
yup 👍

**Comment by tomholub** [comment]
Actually, the whole time we had 2022-05 running, which is the latest version.

Have you tried this smaller PR against a live instance, is it giving you any trouble?

**Comment by tomholub** [comment]
If you test it against a live instance and it works as intended, you can mark it as ready for review. Thanks! 👍

File: extension/chrome/elements/compose-modules/compose-send-btn-module.ts [hunk 1]
```diff
@@ -214,6 +214,9 @@ export class ComposeSendBtnModule extends ViewModule<ComposeView> {
         throw e;
       }
     }
+    if (msg.externalId) {
+      this.view.acctServer.messageGatewayUpdate(msg.externalId, msgSentRes.id).catch(Catch.reportErr);
```
Comment (tomholub) [inline extension/chrome/elements/compose-modules/compose-send-btn-module.ts:218]: I think we can `await` this, and put it just after `draftDelete`. I suppose we could make the two calls in parallel if we want to speed it up a little bit.

**Review by tomholub — COMMENTED** [review]
Looks good - see comment below

**Comment by rrrooommmaaa** [comment]
> Actually, the whole time we had FES 2022-05 running, which is the latest version.
> 
> Have you tried this smaller PR against a live instance, is it giving you any trouble?

I tried that in early February. I don't quite get it -- should I log in as `@flowcrypt.com` to connect to the right instance?

**Comment by tomholub** [comment]
> I tried that in early February. 

ok - that was likely on older instance

> I don't quite get it -- should I log in as @flowcrypt.com to connect to the right instance?

correct - fes.flowcrypt.com will be used on `@flowcrypt.com` account. fes.example.com will be used on example.com account if it's running there, etc.

**Review by tomholub — APPROVED** [review]

**Pull request merged**